itensor/mps/mpoalgs.cc

```diff
@@ -227,24 +227,19 @@ densityMatrixApplyMPOImpl(MPO const& K,
 
     auto res = psi;
 
-    //Set up conjugate psi and K
-    auto psic = psi;
-    auto Kc = K;
-    //TODO: use sim(linkInds), sim(siteInds)
-    psic.dag().prime(rand_plev);
-    Kc.dag().prime(rand_plev);
-
-    // Make sure the original and conjugates match
-    for(auto j : range1(N-1)) 
-        Kc.ref(j).prime(-rand_plev,uniqueSiteIndex(Kc,psic,j));
-
     //Build environment tensors from the left
     if(verbose) print("Building environment tensors...");
     auto E = std::vector<ITensor>(N+1);
-    E[1] = psi(1)*K(1)*Kc(1)*psic(1);
+    auto Kc = dag(prime(K(1),rand_plev));
```

Comment:
Were these changes included by accident?

Reply:
Not accident. Because I found the density matrix apply function used larger memory than expected so I did something to relieve this issue. But this modifications here didn’t fix the problem 100%. Maybe I need to make a separate pull request instead.

Reply:
Yes, please raise an issue and/or make a separate PR for that.